zulip_botserver/tests/test_server.py

```diff
@@ -132,6 +132,64 @@ def test_argument_parsing_defaults(self) -> None:
         assert opts.hostname == '127.0.0.1'
         assert opts.port == 5002
 
+    def test_read_config_from_env_vars(self) -> None:
+        os.environ['ZULIP_BOTSERVER_CONFIG'] = json.dumps({
+            'helloworld': {
+                'email': 'helloworld-bot@zulip.com',
+                'key': 'value',
+                'site': 'http://localhost',
+                'token': 'abcd1234',
+            },
+            'giphy': {
+                'email': 'giphy-bot@zulip.com',
+                'key': 'value2',
+                'site': 'http://localhost',
+                'token': 'abcd1234',
+            }
+        })
+
+        # No bot specified; should read all bot configs
+        bot_conf1 = server.read_config_from_env_vars()
+        expected_config1 = {
+            'helloworld': {
+                'email': 'helloworld-bot@zulip.com',
+                'key': 'value',
+                'site': 'http://localhost',
+                'token': 'abcd1234',
+            },
+            'giphy': {
+                'email': 'giphy-bot@zulip.com',
+                'key': 'value2',
+                'site': 'http://localhost',
+                'token': 'abcd1234',
+            }
+        }
+        assert json.dumps(bot_conf1, sort_keys=True) == json.dumps(expected_config1, sort_keys=True)
```

Comment:
We don't need to serialize these to test equality -- we can just `assert bot_conf1 == expected_config1`

Reply:
Can this stay as it is for consistency?

Reply:
Consistency with what?

Reply:
Oh, I see that `test_read_config_file` also serializes to JSON.  I'd rather fix that, than perpetuate the unnecessary code.